api/enki.py

```diff
@@ -209,17 +210,17 @@ def _make_request(self, url, method, headers, data=None, params=None,
         )
 
     def reaper_request(self, identifier):
-        self.log.debug ("Checking availability for " + str(identifier))
+        self.log.debug ("Checking availability for " + str(identifier.identifier))
         now = datetime.datetime.utcnow()
         url = str(self.base_url) + str(self.item_endpoint)
         args = dict()
         args['method'] = "getItem"
-        args['recordid'] = identifier
+        args['recordid'] = identifier.identifier
         args['size'] = "small"
         args['lib'] = self.lib
         response = self.request(url, method='get', params=args)
         if not(response.content.startswith("{\"result\":{\"id\":\"")):
-            pool = identifier.licensed_through
+            pool = identifier.licensed_through[0]
```

Comment:
`[x for x in identifier.licensed_through if x.data_source==correct_data_source][0]`